services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/update/actions/ValidateFlowAction.java

```diff
@@ -103,14 +103,14 @@ public ValidateFlowAction(PersistenceManager persistenceManager, FlowOperationsD
                 }
             }
 
-            Flow foundFlow = getFlow(flowId, FetchStrategy.NO_RELATIONS);
+            Flow foundFlow = getFlow(flowId, FetchStrategy.DIRECT_RELATIONS);
             if (foundFlow.getStatus() == FlowStatus.IN_PROGRESS) {
                 throw new FlowProcessingException(ErrorType.REQUEST_INVALID,
                         format("Flow %s is in progress now", flowId));
             }
 
-            // Keep it, just in case we have to revert it.
-            stateMachine.setOriginalFlowStatus(foundFlow.getStatus());
```

Comment:
Strictly speaking, the flow object "foundFlow" and the flow entity in neo4j can become different at this point. The opened transaction won't restrain other processes from making changes in the flow. That's why in the original version we locked the flow first and only in the next step (and another transaction) we read and store the flow data.